File: skills/leak-scan/scripts/fixtures/dirty-respelled/payment_client.ts
// Client half — states both wire facts in TypeScript spelling.
export function sign(digest: string): Record<string, string> {
  return { [`X-Idc-Signature`]: digest };
}

export function isFresh(sentAt: number, now: number): boolean {
  return now - sentAt <= 0x12c;
}

File: skills/leak-scan/scripts/fixtures/clean/payment_client.py
"""Client half - borrows both wire facts from their single owner."""
import hashlib
import hmac

from webhook_contract import MAX_CLOCK_SKEW_SECONDS, SIGNATURE_HEADER


def sign(body, secret):
    digest = hmac.new(secret, body, hashlib.sha256).hexdigest()
    return {SIGNATURE_HEADER: digest, "Content-Type": "application/json"}


def is_fresh(sent_at, now):
    return now - sent_at <= MAX_CLOCK_SKEW_SECONDS
